ADALiOS/ADALiOS/ADAuthenticationContext+Internal.m

```diff
@@ -124,16 +125,35 @@ - (ADAuthenticationResult*)updateResult:(ADAuthenticationResult*)result
         return [ADAuthenticationResult resultFromError:error];
     }
     
-    userId = [ADUserInformation normalizeUserId:userId];
-    NSString* actualUser = result.tokenCacheStoreItem.userInformation.userId;
-    if (!userId || AD_SUCCEEDED != result.status || !actualUser)
+    if (AD_SUCCEEDED != result.status || !userId || [NSString adIsStringNilOrBlank:userId.userId] || userId.type == OptionalDisplayableId)
     {
         //No user to compare - either no specific user id requested, or no specific userId obtained:
         return result;
     }
     
+    NSString* normalizedUserId = [ADUserInformation normalizeUserId:userId.userId];
```

Comment:
ADUserIdentifier should have 'id' field not 'userid'. It will look better if we have here:

```
[ADUserInformation normalizeUserId:user.Id];
```